src/include/migraphx/op/unique.hpp

```diff
@@ -0,0 +1,340 @@
+            // for efficiency, loop is bypassed for single elements. ct = 1
+            for(size_t ct = chunk_sz; ct > 1; idx1++, idx2++, ct--)
+            {
+                if(input_data[idx1] < input_data[idx2])
+                    return true;
+                else if(input_data[idx1] > input_data[idx2])
+                    return false;
+            }
+            return input_data[idx1] < input_data[idx2];
+        };
+        std::map<size_t, size_t, decltype(idx_less_fn)> uniq_val_map(idx_less_fn);
+
+        std::vector<std::size_t> y_indices;
+        std::vector<std::size_t> x_rev_indices;
+        std::vector<std::size_t> y_count;
+
+        // go through all the elements and add the unique elements into the map..
+        // inline processing for outputs: y_indices, x_rev_indices, y_count
+        size_t count_x = input_data.size();
+        for(size_t f_idx = 0; f_idx < count_x; f_idx += chunk_sz)
+        {
+            auto [itr, added_new] = uniq_val_map.insert({f_idx, y_indices.size()});
+            if(added_new)
+            {
+                y_count.push_back(0);
+                y_indices.push_back(x_rev_indices.size());
+            }
+            y_count[itr->second]++;
+            x_rev_indices.push_back(itr->second);
+        }
+
+        return std::tuple{y_indices, x_rev_indices, y_count};
+    }
+
+    // Default: none. Range: [-rank, rank-1]
+    static constexpr int64_t axis_none = std::numeric_limits<int64_t>::max();
+    int64_t axis                       = axis_none;
+
+    // Default: 1= sorted. 0 = unsorted.
+    int64_t sorted = 1;
+
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.axis, "axis"), f(self.sorted, "sorted"));
+    }
+
+    std::string name() const { return "unique"; }
+
+    shape compute_shape(std::vector<shape> inputs) const
+    {
+        check_shapes{inputs, *this}.has(1).standard();
+
+        auto& sh_x         = inputs[0];
+        auto lens_x        = sh_x.lens();
+        size_t dim_x       = sh_x.ndim();
+        size_t max_uniq_ct = sh_x.elements();
+
+        int64_t t_axis = axis;
+        if(t_axis != axis_none)
+        {
+            t_axis = migraphx::tune_axis(dim_x, t_axis, name());
+            if(t_axis != 0)
+                MIGRAPHX_THROW("Unique: Only supports axis = 0 or None");
+            max_uniq_ct /= sh_x.strides()[t_axis];
+        }
+
+        // min = 1 unique element; max = full dimension along the axis
+        // The three outputted Indices are just 1-D:
+        std::vector<shape::dynamic_dimension> d_out{{1, max_uniq_ct}};
+        shape sh_idx{shape::int64_type, {d_out}};
+
+        // Unique elements themselves aren't necessarily 1-D.. fix its output shape:
+        for(size_t idx = 1; t_axis != axis_none && t_axis + idx < dim_x; idx++)
+            d_out.push_back({lens_x[t_axis + idx], lens_x[t_axis + idx]});
+        shape sh_y = {sh_x.type(), {d_out}};
```

Comment:
```suggestion
        shape sh_y = {sh_x.type(), d_out};
```

Reply:
We are supplying a vector there.. so isn't the intent more clear this way..? Thanks.

Reply:
But the init list makes it look more like `d_out` is an element of the vector instead of the vector itself, which is why its clearer to write it without the `{}`.

Reply:
You both are right. Sorry. I did change it to a vector somewhere down as the code evolved.. so no need for the extra brace anymore.

Reply:
Done.